### PR TITLE
Add some metrics for cloud replication feed

### PR DESCRIPTION
**Pull request by ankagrawal**
Add metrics to check cosmos change feed cache hit, miss and refresh rate.
Add metrics to check time taken for cloud replication feed query.

### CODE REVIEW

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosChangeFeedBasedReplicationFeed.java [hunk 1]
```diff
@@ -169,49 +170,64 @@ public CosmosChangeFeedBasedReplicationFeed(CosmosDataAccessor cosmosDataAccesso
...
+          changeFeedCacheEntry = getNextChangeFeed(partitionPath, changeFeedCacheEntry.getEndContinuationToken(),
+              changeFeedCacheEntry.getCacheSessionId());
+          fetchedEntries = changeFeedCacheEntry.getFetchedEntries();
+          if (fetchedEntries.isEmpty()) {
+            // this means that there are no new changes
+            break;
+          } else {
+            azureMetrics.changeFeedCacheRefreshRate.mark();
+          }
+          index = 0;
```
Comment (jsjtzyy) [inline ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosChangeFeedBasedReplicationFeed.java:218]: I probably misunderstand this but I am thinking if there is a case where `cacheHit` is true before entering the `while(true)` loop.  The first time code enters in while loop, `index 
 == fetchedEntries.size()` and jumps to `else` branch. It does fetch new changes and index is reset to 0. Then it goes back to beginning of while(true) and `azureMetrics.changeFeedCacheHitRate.mark();` will be called. I just want to confirm this is a possible case and updating `changeFeedCacheHitRate` is what we expect.

Comment (ankagrawal) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosChangeFeedBasedReplicationFeed.java:218]: First time code enters in while loop, and if index == fetchedEntries.size(), then we refresh the cache as you mentioned. In this case we don't want to say cacheHit just yet, because we haven't yet hit the cache. We will refresh the cache and mark it so.

Now once the refresh is successful, there can be either no results (fetchedEntries.isEmpty()), in which case we will come out of loop without marking cache hit (as expected because it isn't technically a cache hit). If there are entries returned  in cache refresh, then loop's next iteration will run, which will mark the cache hit. 

In short unless we are able to consume something from the cache, we don't mark a cache hit. Hope this makes sense.

**Comment by lightningrob** [comment]
Looks like you need to merge one conflict before merging.

File: ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosChangeFeedBasedReplicationFeed.java (hunk 1 above)
Comment (jsjtzyy) [reply ambry-cloud/src/main/java/com/github/ambry/cloud/azure/CosmosChangeFeedBasedReplicationFeed.java:213]: make sense
